File: clients/android/app/src/main/java/local/visual/corepose/PlayerExport.kt
package local.visual.corepose

import java.net.URLEncoder
import java.nio.charset.StandardCharsets

object PlayerExport {
    val SHARE_TARGETS = listOf(
        "YouTube" to "https://www.youtube.com/upload",
        "TikTok" to "https://www.tiktok.com/upload",
        "X" to "https://twitter.com/intent/tweet?text={text}",
        "Facebook" to "https://www.facebook.com/sharer/sharer.php",
        "Weibo" to "https://service.weibo.com/share/share.php?title={text}",
        "Bilibili" to "https://member.bilibili.com/platform/upload/video/frame",
    )

    fun shareUrl(template: String, text: String): String {
        val encoded = URLEncoder.encode(text, StandardCharsets.UTF_8).replace("+", "%20")
        return template.replace("{text}", encoded)
    }

    fun downloadFileName(displayName: String?, clipId: String?): String {
        val raw = displayName?.ifBlank { null } ?: clipId?.ifBlank { null } ?: "overlay"
        val stem = raw.replace(Regex("[\\\\/:*?\"<>|]"), "_")
        return if (stem.endsWith(".jpg", ignoreCase = true)) stem else "$stem.jpg"
    }
}

File: clients/ios/VisualPose/PlayerExport.swift
import Foundation

enum PlayerExport {
    static let shareTargets: [(String, String)] = [
        ("YouTube", "https://www.youtube.com/upload"),
        ("TikTok", "https://www.tiktok.com/upload"),
        ("X", "https://twitter.com/intent/tweet?text={text}"),
        ("Facebook", "https://www.facebook.com/sharer/sharer.php"),
        ("Weibo", "https://service.weibo.com/share/share.php?title={text}"),
        ("Bilibili", "https://member.bilibili.com/platform/upload/video/frame"),
    ]

    static func shareUrl(template: String, text: String) -> String {
        var allowed = CharacterSet.alphanumerics
        allowed.insert(charactersIn: "-_*")
        let encoded = text.addingPercentEncoding(withAllowedCharacters: allowed) ?? text
        return template.replacingOccurrences(of: "{text}", with: encoded)
    }

    static func downloadFileName(displayName: String?, clipId: String?) -> String {
        let raw: String
        if let displayName, !displayName.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
            raw = displayName
        } else if let clipId, !clipId.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
            raw = clipId
        } else {
            raw = "overlay"
        }
        let stem = raw.replacingOccurrences(of: "[\\\\/:*?\"<>|]", with: "_", options: .regularExpression)
        if stem.lowercased().hasSuffix(".jpg") { return stem }
        return "\(stem).jpg"
    }
}
